src/blenderbim/blenderbim/bim/module/pset/operator.py

```diff
@@ -374,3 +374,147 @@ def _execute(self, context):
+            props.remove(self.index)
+        if self.type == "PropertiesToAddOrEdit":
+            props = context.scene.PropertiesToAddOrEdit
+            props.remove(self.index)
+        return {"FINISHED"}
+    
+
+class ClearList(bpy.types.Operator):
+    bl_label = "Clear list of properties"
+    bl_idname = "bim.clear_list"
+    bl_options = {"REGISTER", "UNDO"}
+    type: bpy.props.StringProperty()
+    
+    def execute(self, context):
+        if self.type == "PropertiesToRename":
+            props = context.scene.PropertiesToRename
+            props.clear()
+        if self.type == "PropertiesToAddOrEdit":
+            props = context.scene.PropertiesToAddOrEdit
+            props.clear()
+        return {"FINISHED"}
+    
+
+class RenameParameters(bpy.types.Operator):
+    bl_label = "Rename Parameters"
+    bl_idname = "bim.rename_parameters"
+    bl_options = {"REGISTER", "UNDO"}
+    bl_description = "Rename parameters that are subclasses of IfcElement"
+
+    def execute(self, context):
+        return IfcStore.execute_ifc_operator(self, context)
+
+    def _execute(self, context):
+        props_to_map = context.scene.PropertiesToRename
+        ifc_file = IfcStore.get_file()
+        all_ifc_elements = ifc_file.by_type("IfcElement")
+
+        for ifc_element in all_ifc_elements:
+            for definition in ifc_element.IsDefinedBy:
+                if definition.is_a('IfcRelDefinesByProperties'):
+                    prop_set = definition.RelatingPropertyDefinition
+                    self.rename_property(prop_set, props_to_map, ifc_element)
+
+        self.report({'INFO'}, 'Finished applying changes')
+        return {"FINISHED"}
+                                    
+    def rename_property(self, property_set, properties_to_map, ifc_element):
+        if property_set.is_a() == "IfcPropertySet":
+            property_container = property_set.HasProperties
+        elif property_set.is_a() == "IfcElementQuantity":
+            property_container = property_set.Quantities
+            
+        for obj_prop in property_container:
+            for prop2map in properties_to_map:
+                if prop2map.pset_name != property_set.Name:
+                    continue
+                if prop2map.existing_property_name == obj_prop.Name:
+                    obj_prop.Name = prop2map.new_property_name
+                    Data.load(IfcStore.get_file(), ifc_element.id())    
+
+
+class AddEditCustomProperty(bpy.types.Operator):
+    bl_label = "Add or edit a custom property"
+    bl_idname = "bim.add_edit_custom_property"
+    bl_options = {"REGISTER", "UNDO"}
+    index: bpy.props.IntProperty()
+
+    def execute(self, context):
+        return IfcStore.execute_ifc_operator(self, context)
+
+    def _execute(self, context):
+        self.file = IfcStore.get_file()
+        selected_objects = context.selected_objects
+        props = context.scene.PropertiesToAddOrEdit     
+
+        for object in selected_objects:
```

Comment:
It's good practice to avoid overriding python builtins (`object`,  `type`, `min`, `max`, etc.) to save headaches down the line : https://docs.python.org/3/library/functions.html , that's why you usually see `for obj in context.select_objects` for instance. If you're using a python linter the builtins are usually colored differently so you can differentiate them

Reply:
Agreed.